### PR TITLE
feat: Expose separate setters for map, list and set

### PR DESCRIPTION
**Pull request by jhgg**
There's some cargo fmt noise in here, but...

The `set_collection` function doesn't actually do what the sticker on the box says, namely, it doesn't work with maps & lists. 

This exposes separate setters for each - but I don't know if this is better, since they translate internally to the "same" CFFI calls, we kinda have to expose different calls into rust.

I'm open on thoughts here, one thing we could do is make a trait like `CassCollection` - that would expose the `inner()` function, and be implemented on sets, maps, lists.

Ot there's this. 

Either way it's *probably* a breaking change.

### CODE REVIEW

File: src/cassandra/tuple.rs [hunk 1]
```diff
@@ -151,8 +151,24 @@ impl Tuple {
...
+
+    /// Sets a map in a tuple at the specified index.
+    pub fn set_map<S>(&mut self, index: usize, value: S) -> Result<&mut Self>
+    where
+        S: Into<Map>,
+    {
+        unsafe { cass_tuple_set_collection(self.0, index, value.into().inner()).to_result(self) }
+    }
+
+    /// Sets a set" in a tuple at the specified index.
```
Comment (kw217) [inline src/cassandra/tuple.rs:170]: typo: missing `"`

**Comment by kw217** [comment]
I think this approach is fine, and as breaking changes go it's not a very severe one. The trait might be a bit neater, but it's a faff to use on the client side.

**Review by kw217 — APPROVED** [review]
LGTM.

**Comment by kw217** [comment]
Trivial conflict, and out-of-date. Best to just do one at a time. I'll try and check regularly during 0900-1700 UTC - is there any overlap with your hours?

**Comment by kw217** [comment]
Conflict again. BTW, if you enable "allow edits from maintainers" on your PR, I'll be able to sort this myself. See
https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by jhgg** [comment]
@kw217 no joke - that option literally does not exist for me (i wonder if its because our upstream is an org, vs a user.) 

if it'd make these kinda things easier in the future, you can also give me write access to this repo and I can just submit branches here. 

(as well - would love to help maintain this - as we're using it pretty heavily inside Discord.)

**Pull request merged**